app/services/parent_profile_associator.rb

```diff
@@ -14,18 +14,46 @@ def run!
     private
 
     def find_parent(profile)
-      parents = Profile.where(ref_id: profile.ref_id,
-                              benchmark_id: profile.benchmark_id,
-                              account_id: nil)
-      raise not_found(profile) unless parents.one?
+      logger.info("Finding parents for profile #{profile.id}")
+      parents = find_in_same_benchmark(profile)
+      parents = find_in_other_benchmarks(profile) if parents.blank?
 
-      parents.first
+      raise not_found(profile) unless parents.any?
+
+      parent = find_most_recent_parent(parents)
+      logger.info("Found parent #{parent.id} for profile #{profile.id}")
+      parent
+    end
+
+    def find_in_same_benchmark(profile)
+      Profile.where(ref_id: profile.ref_id,
+                    benchmark_id: profile.benchmark_id,
+                    account_id: nil)
+    end
+
+    # Profiles in the phony-benchmark will have to be assigned to
+    # a real benchmark if it can be found.
+    def find_in_other_benchmarks(profile)
+      Profile.where(name: profile.name,
+                    ref_id: profile.ref_id,
+                    account_id: nil)
+    end
+
+    def find_most_recent_parent(parents)
+      latest_benchmark = ::Xccdf::Benchmark.find_latest(
```

Comment:
Given many prospective parents are found, return the one with the most recent version